tools/check_bazel_package_cycles.py

```
@@ -0,0 +1,128 @@
+
+def package_for_label(label: str) -> str | None:
+    if not any(
+        label == root or label.startswith(f"{root}/")
+        for root in _FIRST_PARTY_ROOTS
+    ):
+        return None
+    return label.split(":", maxsplit=1)[0]
+
+
+def graph_from_query_xml(query_xml: str) -> dict[str, set[str]]:
+    root = ET.fromstring(query_xml)
```

Ruff (0.16.1)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)